internal/cluster/node_reconciler.go

```diff
@@ -0,0 +1,76 @@
+/*
```

Comment:
we should avoid adding reconciler code in `internal` package. It should be in controllers package.